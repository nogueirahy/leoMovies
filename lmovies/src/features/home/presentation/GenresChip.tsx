import React from 'react';

import { View } from 'react-native';
import { Chip } from 'react-native-paper';

import { GenresChipStyle } from './styles';

function GenresChip({ genres = [] }) {
  return (
    <View style={GenresChipStyle.container}>
      {genres.map((item) => (
        <Chip
          key={item.id}
          style={GenresChipStyle.chipContainer}
          textStyle={GenresChipStyle.text}>
          {item.name}
        </Chip>
      ))}
    </View>
  );
}

export default GenresChip;
