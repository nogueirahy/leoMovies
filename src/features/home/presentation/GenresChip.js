import React from 'react';
import PropTypes from 'prop-types';

import { View, StyleSheet } from 'react-native';
import { Chip } from 'react-native-paper';

const propTypes = {
  genres: PropTypes.arrayOf(PropTypes.object),
};

const defaultProps = {
  genres: [],
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingTop: 10,
  },
  chipContainer: {
    margin: 4,
  },
  text: {
    fontStyle: 'italic',
  },
});

function GenresChip({ genres }) {
  return (
    <View style={styles.container}>
      {genres.map(item => (
        <Chip key={item.id} style={styles.chipContainer} textStyle={styles.text}>{item.name}</Chip>
      ))
      }
    </View>
  );
}

GenresChip.propTypes = propTypes;
GenresChip.defaultProps = defaultProps;


export default GenresChip;
