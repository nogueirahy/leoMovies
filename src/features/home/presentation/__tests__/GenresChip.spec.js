import React from 'react';
import { render } from 'react-native-testing-library';

import GenresChip from '../GenresChip';

describe('GenresChip', () => {
  const props = {
    genres: [{
      id: 1,
      name: 'Drama',
    }],
  };

  it('deve renderizar corretamente', () => {
    const tree = render(<GenresChip {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
