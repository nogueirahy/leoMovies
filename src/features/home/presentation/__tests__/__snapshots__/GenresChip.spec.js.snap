// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresChip deve renderizar corretamente 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ebebeb",
        "borderColor": "#ebebeb",
        "borderRadius": 16,
        "borderStyle": "solid",
        "borderWidth": 0.5,
        "elevation": 0,
        "margin": 4,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 16,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 4,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "lineHeight": 24,
                  "marginVertical": 4,
                  "minHeight": 24,
                  "textAlignVertical": "center",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "marginLeft": 8,
                  "marginRight": 8,
                },
                Object {
                  "fontStyle": "italic",
                },
              ],
            ]
          }
        >
          Drama
        </Text>
      </View>
    </View>
  </View>
</View>
`;
